pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -424,3 +431,26 @@ func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1
 	}
 	return nil
 }
+
+func (p *Provider) ResolveClusterCIDR(ctx context.Context) error {
+	if p.ClusterCIDR.Load() != nil {
+		return nil
+	}
+	out, err := p.eksapi.DescribeClusterWithContext(ctx, &eks.DescribeClusterInput{
+		Name: aws.String(options.FromContext(ctx).ClusterName),
+	})
+	if err != nil {
+		return cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving cluster CIDR, %w", err))
+	}
+	if ipv4CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv4Cidr; ipv4CIDR != nil {
+		p.ClusterCIDR.Store(ipv4CIDR)
+		logging.FromContext(ctx).With("cluster-cidr", *ipv4CIDR).Debugf("discovered cluster CIDR")
+		return nil
+	}
+	if ipv6CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv6Cidr; ipv6CIDR != nil {
+		logging.FromContext(ctx).With("cluster-cidr", *ipv6CIDR).Debugf("discovered cluster CIDR")
+		p.ClusterCIDR.Store(ipv6CIDR)
+		return nil
+	}
+	return cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving cluster CIDR, no CIDR found in DescribeCluster response"))
```

Comment:
Since "resolving cluster CIDR" is used here and the other error path, consider just bubbling this up into the parent err handling so that you can just return "no CIDR found in DescribeCluster"

Also, thoughts on just saying "CIDR not found"?
```suggestion
	return cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving cluster CIDR, CIDR not found"))
```

Reply:
Since there are several places cluster details could be discovered from (e.g. options, describe cluster, api server) I think this could be valuable context. @jonathan-innis and I agreed offline to only discover CIDR through describe cluster since it's only needed for those using the AL2023 EKS optimized AMIs.